Design.md

```diff
@@ -0,0 +1,161 @@
+    - Use mTLS authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto
+      setup for certificates. Do not use any other authentication protocols on top of mTLS.
+    - Use a simple authorization scheme.
+- CLI:
+    - CLI should be able to connect to worker service and schedule several jobs.
+    - CLI should be able to query result of the job execution and fetch its output.
+
+## Job object
+
+- `ID string` unique identifier of a job
+- `Command string` Linux command to be executed
+- `Arguments []string` list of arguments for the command
+- `Status string` current status of the job
+- `Output string` output of the job process (combined stderr and stdout)
+- `Error string` startup error, termination error, or exit error
+- `ExitCode integer` exit code of the job process
+
+## Job status
+
+- `Created` Received a job creation request (short-lived since we immediately start the job instead of queueing)
+- `Running` Job process started
+- `Running_Force_Stop` Failed to kill the process after force stop requested
+- `Exited` Job finished successfully
+- `Exited_Error` Job finished with error
+- `Exited_Force_Stop` Job exited after force stop requested
+
+## Job handling
+
+- Start the process using `exec.Cmd` in a separate go-routine
+- Utilize go-channels to communicate with the go-routine
+- The go-routine should exit when the job finishes or timeout reached or force stop requested
+
+## Library
+
+- `CreateJob(command, timeout)`
+    - Generate a unique ID
+    - Initialize a job object using the ID
+    - Save the reference to the job into memory for future access
+    - Start the job in a separate go-routine
+    - Return the job object
+- `StopJob(jobID)`
+    - Fetch the job object from memory by job ID
+    - Get the current status of the job and check if it is running or already stopped
+    - Stop the job process if it is running using `Cmd.Process.Kill()`. However, there can be error while terminating the process and it can be still running. Use `Running_Force_Stop` status to indicate this scenario.
+    - Send confirmation on success or error on termination failure
+- `GetJobStatus(jobID)`
+    - Fetch the job object from memory by job ID
+    - Return the details of a job along with output and error
+
+## API
+
+- `POST /create`
+
+Request:
+```json
+{
+  "command": "bash",
+  "args": ["-c", "pwd"]
+}
+```
+
+Response:
+```json
+{
+  "id": "c2l1qbucie6hpdufnung",
+  "command": "bash",
+  "args": ["-c", "pwd"],
+  "status": "CREATED",
+  "output": "",
+  "error": "",
+  "exitCode": 0
+}
+```
+
+- `GET /status/{id}`
+
+Response:
+```json
+{
+  "id": "c2l1qbucie6hpdufnung",
+  "command": "bash",
+  "args": ["-c", "pwd"],
+  "status": "EXITED",
+  "output": "/Users/das/Downloads/teleport/job-worker\n",
+  "error": "",
+  "exitCode": 0
+}
+```
+
+- `PUT /stop/{id}`
+
+Response:
+```json
+{
+  "message": "Termination requested for job c2l1qbucie6hpdufnung, please check job status"
+}
```

Comment:
nit: this response doesn't seem useful
if `StopJob` succeeds, respond with `200 OK` and no body